### ISSUE TITLE
[Feature Request] Support uploading Media to MongoDB storage 

### ISSUE THREAD

**Issue by devlikepro** [allburov/whatsapp-http-api#508]
If we anyway uses mongodb to save session configuration and authentication, why not saving media there as well? 

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by HuangDaHui** [comment]
Good idea, will the efficiency of requesting this interface be greatly improved? Because downloading media is really slow now
/api/{session}/chats/{chatId}/messages/{messageId}?downloadMedia=true

**Comment by devlikepro** [comment]
Hi! 
We'll add optimization in the `/api/{session}/chats/{chatId}/messages/{messageId}?downloadMedia=true` in `2024.9.1` release, even without mongodb!
Basicly it used to re-download all media, without checking file exists or not, now it won' redownload file if there's already in the filesystem
Just make sure to setup the environment properly to save files between container restart 
https://waha.devlike.pro/docs/how-to/storages/#save-media-files-between-the-container-restarts



[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by HuangDaHui** [comment]
> `/api/{session}/chats/{chatId}/messages/{messageId}?downloadMedia=true`

Hi, I have configured mongodb and set the file cleanup time. Will the files in mongodb be deleted when the time is up?
WHATSAPP_FILES_LIFETIME=10800

**Comment by devlikepro** [comment]
Hi! 
Mongodb doesn't save media files right now, only Local (files) and S3 can be used
https://waha.devlike.pro/docs/how-to/storages/

Yes, when mongodb "media" storage is ready - you can configure `WHATSAPP_FILES_LIFETIME` so it'l remove after that time.

Btw, `WHATSAPP_FILES_LIFETIME` is not supported in S3, you need to take care of it yourself

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by HuangDaHui** [comment]
> Hi! Mongodb doesn't save media files right now, only Local (files) and S3 can be used https://waha.devlike.pro/docs/how-to/storages/
> 
> Yes, when mongodb "media" storage is ready - you can configure `WHATSAPP_FILES_LIFETIME` so it'l remove after that time.
> 
> Btw, `WHATSAPP_FILES_LIFETIME` is not supported in S3, you need to take care of it yourself
> 
> [![patron:PRO](https://camo.githubusercontent.com/4f70c3a9ee2450c288faea5e1a1b7ebd0ffc1571450a59676b300ffb32b32537/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f706174726f6e2d50524f2d313838613432)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

Okay, for now, I still use local, so I will keep cleaning up regularly. I look forward to you updating the mongodb file saving function.

**Comment by devlikepro** [comment]
MongoDB deprecated, I don't think we'll go this way


[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
